spec/spec_helper.rb

```diff
@@ -1,8 +1,132 @@
+require './app/data_mapper_setup'
+require 'sinatra/flash'
+# require './spec/factories/user'
+require 'database_cleaner'
+
+
+# require './app/app'
+
+Capybara.app = Chitterchallenge
+
+# This file was generated by the `rspec --init` command. Conventionally, all
+# specs live under a `spec` directory, which RSpec adds to the `$LOAD_PATH`.
+# The generated `.rspec` file contains `--require spec_helper` which will cause
+# this file to always be loaded, without a need to explicitly require it in any
+# files.
+#
+# Given that it is always loaded, you are encouraged to keep this file as
+# light-weight as possible. Requiring heavyweight dependencies from this file
+# will add to the boot time of your test suite on EVERY test run, even for an
+# individual file that may not need all of that loaded. Instead, consider making
+# a separate helper file that requires the additional dependencies and performs
+# the additional setup, and require it from the spec files that actually need
+# it.
+#
+# The `.rspec` file also contains a few flags that are not defaults but that
+# users commonly want.
+#
+# See http://rubydoc.info/gems/rspec-core/RSpec/Core/Configuration
+RSpec.configure do |config|
+  # config.include FactoryGirl::Syntax::Methods
+
+config.before(:suite) do
+ DatabaseCleaner.strategy = :transaction
+ DatabaseCleaner.clean_with(:truncation)
+end
+
+config.before(:each) do
+ DatabaseCleaner.start
+end
+
+config.after(:each) do
+ DatabaseCleaner.clean
+end
+
+config.include Capybara::DSL
+# rspec-expectations config goes here. You can use an alternate
+# assertion/expectation library such as wrong or the stdlib/minitest
+# assertions if you prefer.
+config.expect_with :rspec do |expectations|
+ # This option will default to `true` in RSpec 4. It makes the `description`
+ # and `failure_message` of custom matchers include text for helper methods
+ # defined using `chain`, e.g.:
+ # be_bigger_than(2).and_smaller_than(4).description
+ #   # => "be bigger than 2 and smaller than 4"
+ # ...rather than:
+ #   # => "be bigger than 2"
+ expectations.include_chain_clauses_in_custom_matcher_descriptions = true
+end
+
+# rspec-mocks config goes here. You can use an alternate test double
+# library (such as bogus or mocha) by changing the `mock_with` option here.
+config.mock_with :rspec do |mocks|
+ # Prevents you from mocking or stubbing a method that does not exist on
+ # a real object. This is generally recommended, and will default to
+ # `true` in RSpec 4.
+ mocks.verify_partial_doubles = true
+end
+
+# The settings below are suggested to provide a good initial experience
+# with RSpec, but feel free to customize to your heart's content.
+
+ # These two settings work together to allow you to limit a spec run
+ # to individual examples or groups you care about by tagging them with
+ # `:focus` metadata. When nothing is tagged with `:focus`, all examples
+ # get run.
+ # config.filter_run :focus
+ # config.run_all_when_everything_filtered = true
+
+#  
```

Comment:
Incorrect indentation detected (column 0 instead of 1).<br>Trailing whitespace detected.